### PR TITLE
David/qubit tracking

### PR DESCRIPTION
**Pull request by DavidSarlle**
This PR adds the parameters need during the qubit flux tracking implemented in qibocal (https://github.com/qiboteam/qibocal/tree/david/qubit_flux_track).

The PR is compatible with all the runcards available in main and qibocal. So we can proceed to merge as soon as the reviewers accept it. 

Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

File: src/qibolab/qubits.py
Comment (stavros11) [inline src/qibolab/qubits.py:52]: Could you please add some docstrings, below each definition, eg.:
```py
brf: float = 0.0
"""Explanation what brf is."""
```
that explain the new fields, particularly `brf` and `ssf_brf` as these may not be clear to everyone (at least they are not to me). Maybe `brf` and `ssf_brf` are also not the best names, as they are not very explanatory, but I understand we want to avoid very long names for usability.

Ideally we should add docstrings to all fields of the `Qubit` as this will improve the docs, but for the purpose of this PR we should at least add to the new fields.

**Review by stavros11 — APPROVED** [review]
Thanks @DavidSarlle. It should be fine to merge this now, but we should keep in mind that the `update` method will be removed when https://github.com/qiboteam/qibocal/pull/526 is merged. @andrea-pasquale please coordinate with @DavidSarlle to properly migrate these updates to the corresponing routines in qibocal.

File: src/qibolab/qubits.py
Comment (AleCandido) [reply src/qibolab/qubits.py:52]: Could you also adopt naming conventions? `Ec` and `Ej` are capitalized as classes (and that's why GitHub is also highlighting them differently).

If you also had a more explicit name for `g` it would be nice.

Comment (stavros11) [reply src/qibolab/qubits.py:52]: > Could you also adopt naming conventions? `Ec` and `Ej` are capitalized as classes (and that's why GitHub is also highlighting them differently).

`T1`, `T2` have the same issue but I guess we made an exception to follow the physics notation.

Comment (DavidSarlle) [reply src/qibolab/qubits.py]: @stavros11 thanks for your comments. I have added docstrings for the new attributes that are not clear. As you said, probably we should add docstrings for all the attributes, because there are some of them that are not clear what they are only with the attribute name.

Comment (DavidSarlle) [reply src/qibolab/qubits.py]: > > Could you also adopt naming conventions? `Ec` and `Ej` are capitalized as classes (and that's why GitHub is also highlighting them differently).
> 
> `T1`, `T2` have the same issue but I guess we made an exception to follow the physics notation.

I was thinking exactly the same. But I have already changed them to small letters. What do you want to do? To me the capital letters are more explanatory because we follow the physics notation... Also g, in superconducting physics is well known that is the readout coupling, so I have added the corresponding docstring but I think is better if we do not change the name

File: src/qibolab/qubits.py
Comment (stavros11) [inline src/qibolab/qubits.py:49]: Is this different than the existing `bare_resonator_frequency` because you are at the sweetspot?

File: src/qibolab/qubits.py
Comment (AleCandido) [reply src/qibolab/qubits.py]: > Special cases aren't special enough to break the rules.
Although practicality beats purity.
https://peps.python.org/pep-0020/

So, I know these exceptions are quite natural, but we do not have clear rules about when it is reasonable to make an exception and when it is not.
It is true that on the one side I am the one introducing the discussion, but my (limited) experience tells me that these issues come over and over again, so it is better just to establish a simple (as possible) rule and follow it.

So:
- if `g` is the readout coupling, I would rather call it `g_readout` or `readout_coupling`, because it might be unambiguous in a paper (or little ambiguous), but in a piece of code we have plenty of variables, so single letter ones are rather confusing
- for `Ec`, `T1` and friends I would use `ec`, `t1` and so on or more telling names as in the case of `g`

However, it is not a priority, the code will work anyhow. I just fear to be a bit more confusing in the long term, and then we won't like to change to avoid breaking compatibility.
Choose what you prefer :)

Comment (DavidSarlle) [reply src/qibolab/qubits.py]: If it is not clear, let's leave the Ec and Ej in capital letters, that is usual physics notation and to me is much more clear

File: src/qibolab/qubits.py
Comment (DavidSarlle) [reply src/qibolab/qubits.py]: @stavros11 you are right. I have modified a little bit the docstring to be more precise

**Comment by andrea-pasquale** [comment]
> It should be fine to merge this now, but we should keep in mind that the `update` method will be removed when [qiboteam/qibocal#526](https://github.com/qiboteam/qibocal/pull/526) is merged. @andrea-pasquale please coordinate with @DavidSarlle to properly migrate these updates to the corresponing routines in qibocal.

I will review soon https://github.com/qiboteam/qibocal/pull/519 and I will eventually open a PR on top of that one to fix the update.

File: src/qibolab/qubits.py
Comment (andrea-pasquale) [inline src/qibolab/qubits.py:48]:
```suggestion
    bare_resonator_frequency_sweetspot
```
Just to improve readability

File: src/qibolab/platform.py
Comment (andrea-pasquale) [inline src/qibolab/platform.py:222]: To avoid creating conflicts with https://github.com/qiboteam/qibolab/pull/596 you can remove those lines.

**Review by andrea-pasquale — COMMENTED** [review]
Thanks @DavidSarlle, the code looks good to me.
I will just remove a few lines to avoid creating conflicts with qiboteam/qibocal#526 and #596. I will port those lines in qibocal.

File: src/qibolab/platform.py
Comment (DavidSarlle) [reply src/qibolab/platform.py:222]: I have removed these lines. @andrea-pasquale let me know when qibocal is ready to update the parameters, otherwise we can not use the flux routines

File: src/qibolab/qubits.py
Comment (DavidSarlle) [reply src/qibolab/qubits.py:48]: modified. Also modified in qibolab_qrc_platforms runcards

File: src/qibolab/platform.py
Comment (andrea-pasquale) [reply src/qibolab/platform.py:222]: I've opened the PR in qibocal https://github.com/qiboteam/qibocal/pull/538 @DavidSarlle

File: src/qibolab/qubits.py
Comment (andrea-pasquale) [inline src/qibolab/qubits.py:48]: @DavidSarlle here can't you use `bare_resonator_frequency` at line 40? Or is it a different quantity?

Comment (andrea-pasquale) [reply src/qibolab/qubits.py:48]: Nevermind I just read the update docstring

Comment (AleCandido) [reply src/qibolab/qubits.py:48]: https://github.com/qiboteam/qibolab/pull/581#discussion_r1332795116

**Review by AleCandido — APPROVED** [review]

**Review by andrea-pasquale — APPROVED** [review]

File: src/qibolab/qubits.py
Comment (stavros11) [inline src/qibolab/qubits.py:48]: @DavidSarlle what is the final decision for this? Here we are still using `brf` but in qiboteam/qibolab_platforms_qrc#54 you changed to `bare_resonator_frequency_sweetspot`.

Comment (andrea-pasquale) [reply src/qibolab/qubits.py:48]: Depending on this I will also update https://github.com/qiboteam/qibocal/pull/538
Personally I think that we should use `bare_resonator_frequency_sweetspot`.
Also `ssf_brf` could be renamed to something like `ratio_sweetspot_qubit_freq_bare_resonator_freq` which is pretty long but at least we know what it is.

Comment (DavidSarlle) [reply src/qibolab/qubits.py]: Sorry @stavros11 and @andrea-pasquale I forgot to push the latest changes where I modified the brf name to bare_resonator_frequency_sweetspot.

@andrea-pasquale regarding the ssf_brf, we can modifiy it as you said, but is a very large name for a variable... I have also included the docstrings with the explanation, so I think is better if we keep it as it is...

Comment (andrea-pasquale) [reply src/qibolab/qubits.py]: I agree, I will not argue for the name of a single variable.
Let's keep `ssf_brf` and merge this PR.

Comment (stavros11) [reply src/qibolab/qubits.py]: Great, if everyone agrees, please merge this when the CI passes.

**Pull request merged**
